tools/utils.py

```diff
@@ -112,6 +113,14 @@ def delete_contents(dirname, exclude=None):
       delete_file(entry)
 
 
+def get_num_cores():
+  # Avoid os.cpu_count() here because it doesn't take into account thread affinity.
+  # We could use os.process_cpu_count() but its only available in 3.13.
+  # multiprocessing.cpu_count() will use os.process_cpu_count() automatically
+  # on 3.13 and above
```

Comment:
Where in the docs do you see that? At I don't see it here IIANM:

https://docs.python.org/3/library/multiprocessing.html#multiprocessing.cpu_count)

Reply:
Crap I read the docs wrong.. updating.